src/foam/core/CSV.js

```diff
@@ -0,0 +1,382 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  refines: 'foam.core.Property',
+
+  properties: [
+    {
+      name: 'fromCSV',
+      value: function fromCSV(value, ctx, prop, csv) {
+        return foam.csv.parse(value, null, ctx);
+      }
+    },
+    {
+      name: 'toCSV',
+      value: function toCSV(value, outputter) { return value; }
+    }
+  ]
+});
+
+/** CSV Outputer. **/
+foam.CLASS({
+  package: 'foam.csv',
+  name: 'Outputer',
+
+  documentation: 'CSV Outputer.',
+
+  properties: [
+    {
+      class: 'String',
+      name: 'buf_',
+      value: ''
+    },
+    {
+      class: 'Boolean',
+      name: 'includeQuotes',
+      help: 'If true, values are quoted with the `"` char',
+      value: true
+    },
+    {
+      class: 'String',
+      name: 'delimiter',
+      value: ','
+    },
+    {
+      class: 'String',
+      name: 'nestedObjectSeperator',
+      value: '__'
+    },
+    {
+      class: 'StringArray',
+      name: 'nestedObjectNames'
+    },
+    {
+      class: 'String',
+      name: 'nlStr',
+      value: '\n'
+    },
+    {
+      class: 'String',
+      name: 'undefinedStr',
+      value: ''
+    },
+    {
+      class: 'Boolean',
+      name: 'outputDefaultValues',
```

Comment:
I don't think this works for CSV, since the location is always present, making it impossible to tell the difference between the default value and an empty string.